### PR TITLE
feat: add endpoints to sdk-instone for password management

### PR DESCRIPTION
**Pull request by roperzh**
This PR is part of a series of efforts destined to enable a password reset workflow, it's ready for review, but please hold on merging this until https://github.com/HaikuTeam/inkstone/pull/18 is merged.

If you want to test this locally, please merge the above mentioned PR in Inkstone and point to your local URL.

[Asana ticket](https://app.asana.com/0/482906470227387/480796620059192)

### CODE REVIEW

File: packages/@haiku/sdk-inkstone/src/index.ts [hunk 1]
```diff
@@ -148,6 +150,44 @@ export namespace inkstone {
...
+
+      const options: requestLib.UrlOptions & requestLib.CoreOptions = {
+        url: inkstoneConfig.baseUrl + ENDPOINTS.RESET_PASSWORD_CLAIM.replace(':UUID', resetPasswordUUID),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({password, passwordConfirmation}),
+      };
+
+      request.put(options, (err, httpResponse, body) => {
```
Comment (stristr) [inline packages/@haiku/sdk-inkstone/src/index.ts]: Nits, but conceptually relevant:

1. `PUT /v0/reset-password/{uuid}` makes less sense than `POST /v0/reset-password/{uuid}` or (arguably more precise) `POST /v0/reset-password/{uuid}/claim` would here. The main reason for this is that `PUT` requests are supposed to be idempotent, but this request is not (it should fail if you call it a second time).
2. Password confirmation matching password is essentially view logic and not model logic, so I wouldn't expect the API to take a second `passwordConfirmation` parameter—that validation should belong to the view and not even the `@haiku/sdk-inkstone` method.

**Comment by stristr** [comment]
LGTM except for RESTful nits above.

File: packages/@haiku/sdk-inkstone/src/index.ts (hunk 1 above)
Comment (roperzh) [reply packages/@haiku/sdk-inkstone/src/index.ts]: @stristr thanks so much for this

**Comment by stristr** [comment]
Merging!

**Comment by roperzh** [comment]
Fixed the RESTful nits both here and on Inkstone

**Comment by stristr** [comment]
@roperzh: if you need to cut a new version of `@haiku/sdk-inkstone` to update the version used by `marketing-website`, you can just [go here](https://ci.haiku.ai/job/Releases/job/Haiku/) and click **Build Now** on the left. It will eventually cut and publish version `2.3.71` of `@haiku/sdk-inkstone` to npm.

**Pull request merged**